tests/fixtures/psub-grid.py

```
@@ -0,0 +1,516 @@
+# performance target, so the ceiling is deliberately far above the observed times.
+import time                                                            # noqa: E402
+
+for _dense in ("su<a>" * 6200, "<(x)" * 7750 + " bash", "su|" * 10000 + " <(printf x)"):
+    n += 1
+    _t0 = time.time()
+    cmdword.is_file_mod(_dense)
+    if time.time() - _t0 > 2.0:
+        bad.append("dense redirect text near the budget cutoff took over 2s")
+
+if bad:
+    print("PSUB fail %d/%d" % (len(bad), n))
```

Ruff (0.16.2)

[warning] Use format specifiers instead of percent format

(UP031)